### DISCUSSION TITLE
How to configure GraphQLContext in kickstart GraphQLHttpServlet as getContext() is deprecated in graphql-java

### DISCUSSION THREAD

**Discussion by David-Java-stack** [graphql-java-kickstart/graphql-java-servlet#438] [Q&A]
Hi, I implemented Instrumentation and using parameters.getContext() to get authentication , however , since now, getContext() is deprecated , I tried to use getGraphQLContext() instead but it returning empty object , what should I do ?
thx

@slf4j
@RequiredArgsConstructor
public class RequestLoggingInstrumentation extends SimpleInstrumentation {

@Override
public InstrumentationContext<ExecutionResult> beginExecution(
        InstrumentationExecutionParameters parameters) {
    GraphQLServletContext context = parameters.getContext();
    HttpServletRequest request = context.getHttpServletRequest();
    KeycloakSecurityContext keycloakSecurityContext = (KeycloakSecurityContext) 
    request.getAttribute(KeycloakSecurityContext.class.getName());

I'm configuring the servlet and configuration as following:

 @Bean
    public ServletRegistrationBean<AbstractGraphQLHttpServlet> partyGraphQLServletRegistration() {
        GraphQLHttpServlet servlet = GraphQLHttpServlet.with(getGraphQLConfiguration());
        return new ServletRegistrationBean<>(servlet, url);
    }
    
     private GraphQLConfiguration getGraphQLConfiguration() {
        return GraphQLConfiguration.with(getSchema()).with(List.of(new LoggingListener())).with(new CustomGraphQLContextBuilder()).with(GraphQLQueryInvoker.newBuilder().withInstrumentation(new RequestLoggingInstrumentation()).build())
                .with(getGraphQLObjectMapper())
                .build();
    }
    
 as u can see I tried to add CustomGraphQLContextBuilder() , however , it appears in getContext() method 
 but getGraphQLContext() is still empty map
 
 any idea? 
 thx

**Comment by oliemansm** [comment]
This was not supported yet in v12. Since v13 you can get the request from the context like this:

```java
public InstrumentationContext beginExecution(InstrumentationExecutionParameters parameters) {
  GraphQlContext context = parameters.getGraphQlContext();
  HttpServletRequest request = context.get(HttpServletRequest.class);
```

**Reply by David-Java-stack** [reply]
thx,  it is working.
